claim/claim.c

```diff
@@ -96,14 +96,19 @@ void load_claiming_state(void)
     // check if the file exists
     if (lstat(filename, &statbuf) != 0) {
         info("File '%s' was not found. Setting state to AGENT_UNCLAIMED.", filename);
-   } else {
-        FILE *f = fopen(filename,"rt");
-        claimed_id = callocz(1,statbuf.st_size+1);
-        size_t bytes_read = fread(claimed_id,1,statbuf.st_size,f);
-        claimed_id[bytes_read] = 0;
-        info("File '%s' was found. Setting state to AGENT_CLAIMED using agent_id=.", filename, claimed_id);
+    } else {
+        if (likely(statbuf.st_size)) {
+            FILE *f = fopen(filename, "rt");
+            if (likely(f)) {
+                claimed_id = callocz(1, statbuf.st_size + 1);
+                size_t bytes_read = fread(claimed_id, 1, statbuf.st_size, f);
+                claimed_id[bytes_read] = 0;
+                info("File '%s' was found. Setting state to AGENT_CLAIMED", filename);
+                fclose(f);
+            } else
+                info("File '%s' was found. Setting state to AGENT_CLAIMED", filename);
```

Comment:
err msg looks to have small copy pasta error + i would log also errno/strerror

Reply:
Fixed error and use error() that logs the errno and message as well